### ISSUE TITLE
moveCamera() doesn't work in onMapCreated callback

### ISSUE THREAD

**Issue by l2-map-dev** [m0nac0/flutter-mapbox-gl#371]
I want to initialize a map with an initial bounding box, and since `initialCameraPosition` appears to only support specifying a center position and zoom level, it appears the appropriate way to accomplish this is by calling `moveCamera()` in the `onMapCreated` callback like so:

```
onMapCreated: (mapController) {
  mapController
      .moveCamera(
    CameraUpdate.newLatLngBounds(newBounds),
}
```

However, this call to moveCamera doesn't seem to do anything; the camera position remains at whatever the initial camera position was set to when creating the widget.

Sleeping with a future before running the update resolves the issue, but this is obviously hacky and I'd like to find a more reliable solution.

```
onMapCreated: (mapController) {

Future.delayed(const Duration(milliseconds: 200), () {
  mapController
      .moveCamera(
    CameraUpdate.newLatLngBounds(newBounds),
})
}
```

I've also tried using 

```
WidgetsBinding.instance.addPostFrameCallback((_) {

mapController
      .moveCamera(
    CameraUpdate.newLatLngBounds(newBounds)

});
```

to delay calling `moveCamera()` until after the widget has been initialized, and oddly, using this method, the camera moves partially to the new position, but stops before reaching the new bounding box.

I've tried all of the above in `onStyleLoaded()` as well with no luck. It seems as though even at the point `onMapCreated()` or `onStyleLoaded()` are run, some state still isn't initialized which prevents the camera update from functioning. Can anyone offer some help here?

**Comment by m0nac0** [comment]
Can you try the onStyleLoadedCallback, please?

**Comment by l2-map-dev** [comment]
Thanks for the reply! As I mentioned above, I did give that a try and unfortunately it had the same result. Even in the `onStyleLoadedCallback`, the camera remains at its initial position. I used `mapController.getVisibleRegion()` to verify this and the bounds of the map are identical before and after the call to `moveCamera`.

**Comment by LouisRaverdy** [comment]
This issue specifically affects iOS devices, while it functions perfectly on Android devices. The problem originates from the onMapCreated and onStyleLoaded callbacks, which are triggered prematurely on iOS. The question is, what is not being waited for? Maybe @josxha, do you have an idea ?

**Comment by josxha** [comment]
Apparently, `onMapCreated()` and `onStyleLoaded ` aren't save to call operations on the map from on iOS. I haven't looked at the at the ios implementation in depth. On android the [onMapCreated()](https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.maps/-map-fragment/on-map-ready.html) callback get's awaited internally to ensure the map has loaded. This callback does not exist in the iOS SDK. There seems to be some code in swift to work around this but it's currently not clear to me why it is done the way it is. 

https://github.com/maplibre/flutter-maplibre-gl/blob/5a3daa23a46cd14b97c4348f2295c97f7a151086/maplibre_gl/ios/maplibre_gl/Sources/maplibre_gl/MapLibreMapController.swift#L130-L143

**Comment by LouisRaverdy** [comment]
I see now, thanks for the clarification. The problem is that there isn't a proper solution, which is really frustrating. The only workaround so far is to add a delay.

**Comment by josxha** [comment]
Yep, I agree. Have you tried to enforce the viewport of the map with the `cameraTargetBounds` parameter and changing the bounds afterwards to unbound? (Problably again after a delay)
